src/AllowList.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import {IAttestation} from "./interfaces/IAttestation.sol";
+import {QuoteParser, WorkloadId} from "./utils/QuoteParser.sol";
+import {TD10ReportBody} from "automata-dcap-attestation/contracts/types/V4Structs.sol";
+
+// TEE identity and status tracking
+struct RegisteredTEE {
+    uint64 registeredAt; // The most recent timestamp that the TEE last registered with a valid quote
+    WorkloadId workloadId; // The workloadID of the TEE device
+    bytes rawQuote; // The raw quote from the TEE device, which is stored to allow for future quote re-verification
+}
+
+/**
+ * @title AllowList
+ * @dev A contract for managing trusted execution environment (TEE) identities and configurations
+ * using Automata's Intel DCAP attestation
+ */
+contract AllowList {
+    // Constants
+
+    // Maximum size for byte arrays to prevent DoS attacks
+    uint256 public constant MAX_BYTES_SIZE = 20 * 1024; // 20KB limit
+
+    // Structs
+
+    // Storage Variables
+
+    // The address of the Automata DCAP Attestation contract, which verifies TEE quotes.
+    // This is deployed by Automata, and once set on the AllowList, it cannot be changed
+    IAttestation public attestationContract;
+
+    // Used to track registered TEEs by the Ethereum address in the quote that originally registered them
+    mapping(address => RegisteredTEE) public registeredTEEs;
+
+    // Used to track registered TEEs by their (ethAddress, workloadId) pair,
+    // which is needed for when
+    mapping(WorkloadId => RegisteredTEE) public registeredTEEsByWorkloadId;
```

Comment:
not sure if we need it? besides, this isn't a 1:1 mapping, as multiple instances can exist with the same workloadid but with different teeAddresses